### DISCUSSION TITLE
How to run Theia portable?

### DISCUSSION THREAD

**Discussion by XeonG** [eclipse-theia/theia#14508] [General] [answered]
Would be useful, sick of trying to backup all the different damn locations of vscode stuff

**Comment by msujew** [comment]
Hey @XeonG,

you can make Theia portable by creating a `data` folder in the application directory. See also:

https://github.com/eclipse-theia/theia/blob/2a0232d74807f861d1cc0df499359e3f2d5362b5/packages/core/src/electron-main/electron-main-application.ts#L199-L211

**Comment by XeonG** [comment]
yeah but the installer could really do with an option inside of it to let you specify the installation path and a tick box to say this is portable so it just does that.. 

here I used the installer and added the 'data' folder.. run theia.. empty folder 
![explorer_tbHui2XRso](https://github.com/user-attachments/assets/1df58a97-db92-4603-96fe-094dd219f25f)

instructions unclear.. its not even documented on the site from what I could see

**Reply by msujew** [reply]
It might make sense to create an issue in the respective repos for that, i.e. https://github.com/eclipse-theia/theia-ide or https://github.com/eclipse-theia/theia-website.

**Reply by cristiandmonda** [reply, marked as answer]
IIRC the `data` folder needs to be placed in `/resources/app` within the application directory to be picked up.

**Reply by XeonG** [reply]
that seems to work.. surprised its not clearly documented or even on the main website with portable version download setup for it

**Reply by msujew** [reply]
> surprised its not clearly documented or even on the main website with portable version download setup for it

Please go ahead and create an issue in the respective repos for that, i.e. https://github.com/eclipse-theia/theia-ide or https://github.com/eclipse-theia/theia-website. :)
